src/main/java/com/github/cameltooling/lsp/internal/completion/CamelOptionValuesCompletionsFuture.java

```diff
@@ -88,6 +89,22 @@ public List<CompletionItem> apply(CamelCatalog camelCatalog) {
 					items.add(langItem);
 				}
 				return items;
+			} else if(StringUtils.startsWithIgnoreCase(
+					optionParamValueURIInstance.getOptionParamURIInstance().getValue().getValueName(),
+					"{{secret:")) {
+				List<CompletionItem> items = new ArrayList<>();
+				try (KubernetesClient client = KubernetesConfigManager.getInstance().getClient()) {
+					var secrets = ((KubernetesClientImpl) client).inAnyNamespace().secrets().list().getItems();
+					secrets.forEach(secret-> secret.getData().forEach((k, v) -> {
+						CompletionItem langItem =
+								new CompletionItem("{{secret:" + secret.getMetadata().getName() + "/" + k + "}}");
+						CompletionResolverUtils.applyTextEditToCompletionItem(optionParamValueURIInstance, langItem);
+						items.add(langItem);
+					}));
```

Comment:
to avoid a cast to an implementation and a potential CalssCast Exception, I think we need to check that it is a NamespacedKubernetesClient. (I didn't know there can be non-namespaced ones), something like:
```suggestion
					if (client instanceof NamespacedKubernetesClient nsClient) {
						var secrets = nsClient.inAnyNamespace().secrets().list().getItems();
						secrets.forEach(secret -> secret.getData().forEach((k, v) -> {
							CompletionItem langItem = new CompletionItem(
									"{{secret:" + secret.getMetadata().getName() + "/" + k + "}}");
							CompletionResolverUtils.applyTextEditToCompletionItem(optionParamValueURIInstance,
									langItem);
							items.add(langItem);
						}));
					}
```